File: beginner/1037/Interval.java
// challenge link - https://judge.beecrowd.com/en/problems/view/1037

import java.io.IOException;
import java.util.Scanner;
import java.lang.Math;

public class Interval {
	public static void main(String[] args) throws IOException {
		double value;
		Scanner reader;
		String interval = "";

		reader = new Scanner(System.in);
		value = reader.nextDouble();
		reader.close();

		if (value < 0 || value > 100) {
			System.out.println("Fora de intervalo");
			return;
		}

		if (value >= 0 && value <= 25) {
			interval = "[0,25]";
		} else if (value > 25 && value <= 50) {
			interval = "(25,50]";
		} else if (value > 50 && value <= 75) {
			interval = "(50,75]";
		} else {
			interval = "(75,100]";
		}

		System.out.println("Intervalo " + interval);
	}
}

File: beginner/1037/interval.c
// challenge link - 

#include <stdio.h>

int main(void) 
{
	int intervals[] = { 0, 25, 50, 75, 100 };
	int intervals_size = 5;
	double value = 0;

	scanf("%lf", &value);
	
	int start = intervals[0];
	int end = intervals[intervals_size -1];

	if (value < start || value > end) {
		printf("Fora de intervalo\n");
		return (0);
	}

	end = intervals[1];

	if (value >= start && value <= end) {
		printf("Interval [%d,%d]\n", start, end);
		return (0);
	}

	for (int i = 1, j = 2; j < intervals_size; i++, j++) {
		start = intervals[i];
		end = intervals[j];

		if (value > start && value <= end) {
			printf("Interval (%d,%d]\n", start, end);
			break;
		}
	}

	return (0);
}
